app/models/alchemy/element.rb

```diff
@@ -23,28 +23,65 @@ class Element < ActiveRecord::Base
     include Alchemy::Touching
     include Alchemy::Hints
 
-    FORBIDDEN_DEFINITION_ATTRIBUTES = %w(contents available_contents amount picture_gallery taggable hint)
-    SKIPPED_ATTRIBUTES_ON_COPY = %w(id position folded created_at updated_at creator_id updater_id cached_tag_list)
+    FORBIDDEN_DEFINITION_ATTRIBUTES = [
+      "amount",
+      "available_contents",
+      "nestable_elements",
+      "contents",
+      "hint",
+      "picture_gallery",
+      "taggable"
+    ].freeze
+
+    SKIPPED_ATTRIBUTES_ON_COPY = [
+      "cached_tag_list",
+      "created_at",
+      "creator_id",
+      "id",
+      "folded",
+      "position",
+      "updated_at",
+      "updater_id"
+    ].freeze
 
     acts_as_taggable
 
-    # All Elements inside a cell are a list. All Elements not in cell are in the cell_id.nil list.
-    acts_as_list scope: [:page_id, :cell_id]
+    # All Elements that share the same page id, cell id and parent element id are considered a list.
+    #
+    # If cell id and parent element id are nil (typical case for a simple page),
+    # then all elements on that page are still in one list,
+    # because acts_as_list correctly creates this statement:
+    #
+    #   WHERE page_id = 1 and cell_id = NULL AND parent_element_id = NULL
+    #
+    acts_as_list scope: [:page_id, :cell_id, :parent_element_id]
+
     stampable stamper_class_name: Alchemy.user_class_name
 
     has_many :contents, -> { order(:position) }, dependent: :destroy
+
+    # Elements can have other elements nested inside
+    has_many :nested_elements,
+      -> { order(:position).not_trashed },
+      class_name: 'Alchemy::Element',
+      foreign_key: :parent_element_id
+
     belongs_to :cell
     belongs_to :page
+
+    # A nested element belongs to a parent element.
+    belongs_to :parent_element, class_name: 'Alchemy::Element', touch: true
+
     has_and_belongs_to_many :touchable_pages, -> { uniq },
       class_name: 'Alchemy::Page',
       join_table: ElementToPage.table_name
 
-    validates_presence_of :name, :on => :create
-    validates_format_of :name, :on => :create, :with => /\A[a-z0-9_-]+\z/
+    validates_presence_of :name, on: :create
```

Comment:
Rails/Validation: Prefer the new style validations `validates :column, presence: value` over `validates_presence_of`.